ironfish/src/primitives/rawTransaction.ts

```diff
@@ -264,7 +270,15 @@ export class RawTransactionSerde {
 
     const version = reader.readU8()
 
-    const raw = new RawTransaction(version)
+    if (version !== RawTransactionSerde.VERSION) {
+      throw new RawTransactionSerdeError(
+        `RawTransaction serialization version ${version} does not match current version ${RawTransactionSerde.VERSION}`,
+      )
+    }
```

Comment:
Alternatively, we could still try to deserialize if `version` is lower than the minimum version and use the version we read here in place of `transactionVersion`